### ISSUE TITLE
no longer necessary twisted.web.test.test_agent.AgentTests.completeConnection should be removed

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6701]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#6701|
|Type|enhancement|
|Created|2013-08-26 12:41:39Z|
|Branch|https://github.com/twisted/twisted/tree/connection-completes-itself-6701|

`Agent` was re-implemented in top of endpoints.  The `reactor.advance(0)` that was necessary when using `ClientCreator` is no longer necessary.

<details><summary>Searchable metadata</summary>

```
trac-id__6701 6701
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_connection_completes_itself_6701 branches/connection-completes-itself-6701
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__web web
keywords__tests tests
time__1377520899000000 1377520899000000
changetime__1377523945000000 1377523945000000
version__None None
owner__exarkun exarkun
cc__jknight
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [39739]) Branching to 'connection-completes-itself-6701'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve set owner to @exarkun|
|-|-|

Sounds reasonable, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [39742]) Merge connection-completes-itself-6701

Author: exarkun
Reviewer: therve
Fixes: [#6701](https://github.com/twisted/twisted/issues/6701)

Remove some testing code that is no longer needed.
